tests/std/tests/P0355R7_calendars_and_time_zones_time_zones/test.cpp

```diff
@@ -91,9 +117,268 @@ void timezone_names_test() {
     assert(_Locate_zone_impl(tzdb.zones, "America/Nuuk") == nullptr); // does not exist in ICU (very rare)
 }
 
+void validate_timezone_transitions(const time_zone* tz, const Transition& data) {
+    auto info = tz->get_info(data.begin());
+    assert(info.begin == data.begin());
+    assert(info.end == data.end());
+    assert(info.offset == data.offset());
+    assert(info.save == data.save());
+    assert(info.abbrev == data.abbrev());
+}
+
+void timezone_sys_info_test() {
+    const auto& tzdb = get_tzdb();
+
+    auto sydney_tz = tzdb.locate_zone(Sydney::Tz_name);
+    assert(sydney_tz != nullptr);
+    validate_timezone_transitions(sydney_tz, Sydney::Day_1);
+    validate_timezone_transitions(sydney_tz, Sydney::Std_1);
+    validate_timezone_transitions(sydney_tz, Sydney::Day_2);
+
+    auto la_tz = tzdb.locate_zone(LA::Tz_name);
+    assert(la_tz != nullptr);
+    validate_timezone_transitions(la_tz, LA::Day_1);
+    validate_timezone_transitions(la_tz, LA::Std_1);
+    validate_timezone_transitions(la_tz, LA::Day_2);
+
+    auto begin  = sydney_tz->get_info(Sydney::Std_1.begin());
+    auto middle = sydney_tz->get_info(Sydney::Std_1.begin() + days{1});
+    auto end    = sydney_tz->get_info(Sydney::Std_1.end());

```

Comment:
`begin` and `end` are the names of functions in `std`; I recommend choosing different names.